[material-ui][docs] Categorize the "How-to guides" pages

danilo-leal opened the pull request:
<!-- Thanks so much for your PR, your contribution is appreciated! ❤️ -->

- [x] I have followed (at least) the [PR section of the contributing guide](https://github.com/mui/material-ui/blob/HEAD/CONTRIBUTING.md#sending-a-pull-request).

This PR attempts to add some categorization to the Material UI "how-to guides" pages. I've felt they needed something like this for a while, as there are already multiple pages with different purposes. 

https://deploy-preview-40403--material-ui.netlify.app/material-ui/guides/material-3-components/

oliviertassinari commented:
It reminds me a bit of this docs reorganization discussion: https://github.com/mui/material-ui/pull/39972#discussion_r1404378529.

DiegoAndai commented:
What do you think about:

- Moving the customization ones to the customization section
- Moving the integrations to a new integrations section
- Moving "Migration from @material-ui/pickers" into the migration section
- Having the crucial guides without a subsection and a "General" section for more overview stuff:

<img width="268" alt="Screenshot 2024-01-03 at 10 58 39" src="https://github.com/mui/material-ui/assets/16889233/5f4b426a-a05e-47ae-8a20-ee4e45001f69">

danilo-leal commented:
Yeah, I was hesitant to move things around _too much_ on this PR for the sake of not having SEO impact, but your suggestion makes sense, and as I also mentioned in the description, there are generally more sorting improvement opportunities! Here's an iteration I'm playing with after your comment, and it's making sense to me, but there would be an impact on SEO as the path changes for some of these pages. Let me know what y'all think about this:

| Customization | How-to guides | Migration | Integrations |
|--------|--------|--------|--------|
| <img width="294" alt="Screenshot 2024-01-03 at 12 12 40" src="https://github.com/mui/material-ui/assets/67129314/19f57ba2-0618-43ce-9be8-ad7c6ded776e"> | <img width="294" alt="Screenshot 2024-01-03 at 12 12 50" src="https://github.com/mui/material-ui/assets/67129314/336b7dd4-7cd3-4535-a82d-691c9c4d9bb1"> | <img width="294" alt="Screenshot 2024-01-03 at 12 13 13" src="https://github.com/mui/material-ui/assets/67129314/04ebc7a8-02ff-4267-a872-98dca32034fa"> | <img width="294" alt="Screenshot 2024-01-03 at 12 13 03" src="https://github.com/mui/material-ui/assets/67129314/62dfd5b7-e2a5-4943-bfa0-87a04f5e43a3"> | 

(There's also an opportunity to improve the theme-related pages there — the "Theming", "Creating themed components", and "Components" pages are all confusing as in they _seem_ to be talking about very similar things, mainly the last two. Similarly with "Palette" and "Color").

DiegoAndai commented:
> I was hesitant to move things around too much on this PR for the sake of not having SEO impact

I cannot contribute to the SEO discussion as I have minimal experience.

The proposal makes sense from an information organization structure, though 👍🏼.

I propose a more significant change as I would like to fix this issue once and for all rather than return to it in 6 months.

danilo-leal commented:
@oliviertassinari —  any thoughts here? Maybe a more SEO-oriented perspective?

danilo-leal commented:
@oliviertassinari — additional bump here! Let me know your thoughts on the proposal at https://github.com/mui/material-ui/pull/40403#issuecomment-1875540979 and potential SEO consequences!

oliviertassinari commented:
@danilo-leal On the SEO front, I'm not aware of any correlation between the link orders in the page and SEO. The changes look independent.

Otherwise, the grouping makes sense. It says simple but with a bit more order (not too much).

oliviertassinari commented:
@danilo-leal On the SEO front, I'm not aware of any correlation between the link orders in the page and SEO. The changes look independent.

Otherwise, the grouping makes sense. It says simple but with a bit more order (not too much).

danilo-leal commented:
Okay, great — thanks for the feedback! I just committed the proposal I posted on https://github.com/mui/material-ui/pull/40403#issuecomment-1875540979, and they might mean some URL changes, which I assume affect SEO to some extent, but as you said, these aren't necessarily "SEO-heavy" pages, and there wouldn't be a lot of changes, too!

> Where would the Framework content would go in the future (guide content type)

I would intuitively think they belong in the Integrations section—that is, if we want to have specific pages for each "framework" (if that applies to us), as we've recently done with Next.js.

danilo-leal commented:
Latest changes on the commit above include:

- Removed the "General" subsection on the How-to guides. I felt like I was mostly adding it due to the "Understanding MUI packages" link, which I ended up moving to the Getting started section as it feels like something worth knowing to start off with proper context.
- Also removed the "General" subsection on the Customization section, and just moved the pages that felt broader to the top.

danilo-leal commented:
Bumping this one up — any thoughts/concerns? Would love to get a review :)

oliviertassinari commented on docs/data/material/pages.ts:
```diff
@@ -158,64 +162,81 @@ const pages: MuiPage[] = [
-      { pathname: '/material-ui/guides/theme-scoping' },
       { pathname: '/material-ui/guides/minimizing-bundle-size' },
-      { pathname: '/material-ui/guides/composition' },
       { pathname: '/material-ui/guides/routing' },
```
To match the page title: https://mui.com/material-ui/guides/routing/.

```suggestion
      { pathname: '/material-ui/guides/routing', title: 'Routing libraries' },
```

oliviertassinari replied:
Also, this page is about how to integration with the link component of frameworks, so to move under integration?

oliviertassinari commented on docs/data/material/pages.ts:
```diff
@@ -158,64 +162,81 @@ const pages: MuiPage[] = [
+        pathname: '/material-ui/integrations/interoperability',
+        title: 'Style library interoperability',
+      },
+      { pathname: '/material-ui/integrations/theme-scoping' },
```
This page is weird. I would expect the content to be split into 3 different pages:

1. Integration > Usage with styled-components
2. Integration > Usage with emotion
3. How-to guide > Theme scoping

On the point 3. I'm not so sure, really depends on either this feature is going away as soon as the runtime-less CSS-in-JS engine adoption reaches > 50% or if there will still be a use case for it. I assumed this feature will go away, so maybe we don't want 3.

danilo-leal replied:
I'd agree that it's a weird page! Will leave further improvements to it for a separate/dedicated PR, though!

oliviertassinari commented on docs/data/material/pages.ts:
```diff
@@ -297,6 +310,18 @@ const pages: MuiPage[] = [
+  {
+    pathname: '/material-ui/discover-more',
+    children: [
+      { pathname: '/material-ui/discover-more/showcase' },
```
**Off-topic**

The showcase is so buried in the content that it feels it defeats the purpose of the page. I think https://nextjs.org/ has it right, it's one of the first things we want new users to check to build up trust with them.

<img src="https://github.com/mui/material-ui/assets/3165635/e8d05899-a62d-436f-98ca-cd979691630d" width="682">

danilo-leal replied:
Agreed. Their structure is a bit different, though (i.e., marketing site links persist in the navbar when transitioning to the documentation). It's a perk of having a domain per product, in my perspective. In any case, maybe the Showcases page could be moved over to the Getting started. There might be an argument to be made that quickly seeing projects developed with the library helps people make a decision and commit to it.

oliviertassinari replied:
I was thinking of having the showcase link visible from https://mui.com/material-ui/ page, maybe like done here:

![Screenshot 2024-01-16 at 21 35 05](https://github.com/mui/material-ui/assets/3165635/99197344-7e20-44dd-a050-115b11b58cb7)

oliviertassinari reviewed (commented):
I imagine we left the links broken before the structure is agreed on to avoid double work, to fix before merging though.

oliviertassinari reviewed (commented):
The structure makes sense to me👍

danilo-leal commented:
Not fully sure about how to fix the failed text_regressions check... would appreciate it if someone could give a hand 😃

DiegoAndai commented:
@danilo-leal did you try rerunning the regressions test already? I haven't encountered this issue before.

danilo-leal commented:
Yeah, just did and got the same error (something with the right-to-left demos) 🤔

danilo-leal commented:
I have no clue why the regressions test is failing 😅

oliviertassinari reviewed (changes requested):
_headers

danilo-leal commented:
@oliviertassinari any clearer pointers about what you mean with "_headers"? I don't think I have to do anything on the `_headers` file, right? 🤔

danilo-leal commented:
I've done that on the commit above, but the regressions test is failing the same way (and it seems to be something unrelated to the PR?).

michaldudak commented:
The regression test fail because of the changed path. These tests are configured to ignore demos from "Guides" directory, but since the RTL page was moved outside of it, tests try to run it. The failing demo requires some additional setup the test runner does not provide, so let's add it back to the ignored list (test/regressions/index.js, the `blacklist` array at line 29).

danilo-leal commented:
Okay — regression test is sorted out; thank you, @michaldudak! But the Vale check can't just be happy now, and I don't know if my last attempt there should be the way to go. I don't know why it started to fail midway through this PR, given I was not changing any content within the moved files. 😪

alexfauquette commented:
I've made vale as not failing is case of error

@oliviertassinari I know you're in favor of having a PR failing to force devs to fix it and avoid accumulating mistakes. But here it was failing just because for more that 50 warning the action adding message fails. So it's the only solution I can think about (except solving all the warning in the files @danilo-leal moved

danilo-leal commented:
Woo, thank you @alexfauquette! I definitely think we should resolve all of the Vale errors that were being picked up here, but that also means revising the copywriting for several pages, which is a big endeavor, to unblock this PR. It doesn't feel like these things need to be that tied together :) @oliviertassinari would love a double-check here from you.

oliviertassinari commented on docs/data/material/customization/theming/theming.md:
```diff
@@ -11,14 +11,14 @@ To promote greater consistency between apps, light and dark theme types are avai
 
 If you wish to customize the theme, you need to use the `ThemeProvider` component in order to inject a theme into your application.
-However, this is optional; Material UI components come with a default theme.
+However, this is optional; Material&nbsp;UI components come with a default theme.
```
Unbreakable space, not regular space

```suggestion
However, this is optional; Material UI components come with a default theme.
```

oliviertassinari commented on docs/data/material/customization/theming/theming.md:
```diff
@@ -11,14 +11,14 @@ To promote greater consistency between apps, light and dark theme types are avai
 ## Theme configuration variables
 
-Changing the theme configuration variables is the most effective way to match Material UI to your needs.
+Changing the theme configuration variables is the most effective way to match Material&nbsp;UI to your needs.
```
Unbreakable space, not regular space

```suggestion
Changing the theme configuration variables is the most effective way to match Material UI to your needs.
```

oliviertassinari commented on docs/data/material/customization/theming/theming.md:
```diff
@@ -33,7 +33,7 @@ You can check out the [default theme section](/material-ui/customization/default
 ### Custom variables
 
-When using Material UI's theme with [MUI System](/system/getting-started/) or [any other styling solution](/material-ui/guides/interoperability/), it can be convenient to add additional variables to the theme so you can use them everywhere.
+When using Material&nbsp;UI's theme with [MUI&nbsp;System](/system/getting-started/) or [any other styling solution](/material-ui/integrations/interoperability/), it can be convenient to add additional variables to the theme so you can use them everywhere.
```
Unbreakable space, not regular space

```suggestion
When using Material UI's theme with [MUI System](/system/getting-started/) or [any other styling solution](/material-ui/integrations/interoperability/), it can be convenient to add additional variables to the theme so you can use them everywhere.
```

danilo-leal replied:
Thank you for fixing all of them! 👍

oliviertassinari commented on docs/data/material/customization/theming/theming.md:
```diff
@@ -87,7 +87,7 @@ To add extra variables to the `theme.palette`, see [palette customization](/mate
 The community has built great tools to build a theme:
 
-- [mui-theme-creator](https://zenoo.github.io/mui-theme-creator/): A tool to help design and customize themes for the Material UI component library. Includes basic site templates to show various components and how they are affected by the theme
+- [mui-theme-creator](https://zenoo.github.io/mui-theme-creator/): A tool to help design and customize themes for the Material&nbsp;UI component library. Includes basic site templates to show various components and how they are affected by the theme
```
Unbreakable space, not regular space

```suggestion
- [mui-theme-creator](https://zenoo.github.io/mui-theme-creator/): A tool to help design and customize themes for the Material UI component library. Includes basic site templates to show various components and how they are affected by the theme
```

oliviertassinari commented on docs/data/material/pages.ts:
```diff
@@ -15,6 +15,10 @@ const pages: MuiPage[] = [
       { pathname: '/material-ui/getting-started/learn' },
       { pathname: '/material-ui/getting-started/design-resources' },
       { pathname: '/material-ui/getting-started/faq', title: 'FAQs' },
+      {
+        pathname: '/material-ui/getting-started/understand-mui-packages',
+        title: 'Understanding MUI packages',
+      },
```
This pages doesn't exist 

https://deploy-preview-40403--material-ui.netlify.app/material-ui/getting-started/faq/

danilo-leal replied:
Tackled this in https://github.com/mui/material-ui/pull/40988 👍
